mne/io/curry/curry.py

```diff
@@ -191,39 +225,76 @@ def _read_curry_info(curry_paths):
 
     all_chans = list()
     for key in ["meg", "eeg", "misc"]:
+        chanidx_is_explicit = (len(curry_params.chanidx_in_file["CHAN_IN_FILE"
+                                   + CHANTYPES[key]]) > 0)    # channel index
+        # position in the datafile may or may not be explicitly declared,
+        # based on the CHAN_IN_FILE section in info file
         for ind, chan in enumerate(labels["LABELS" + CHANTYPES[key]]):
-            ch = {"ch_name": chan,
-                  "unit": curry_params.unit_dict[key],
-                  "kind": FIFFV_CHANTYPES[key],
-                  "coil_type": FIFFV_COILTYPES[key],
-                  }
-            if key == "eeg":
-                loc = np.array(sensors["SENSORS" + CHANTYPES[key]][ind], float)
-                # XXX just the sensor, where is ref (next 3)?
-                assert loc.shape == (3,)
-                loc /= 1000.  # to meters
-                loc = np.concatenate([loc, np.zeros(9)])
-                ch['loc'] = loc
-                # XXX need to check/ensure this
-                ch['coord_frame'] = FIFF.FIFFV_COORD_HEAD
-            elif key == 'meg':
-                pos = np.array(sensors["SENSORS" + CHANTYPES[key]][ind], float)
-                pos /= 1000.  # to meters
-                pos = pos[:3]  # just the inner coil
-                pos = apply_trans(curry_dev_dev_t, pos)
-                nn = np.array(normals["NORMALS" + CHANTYPES[key]][ind], float)
-                assert np.isclose(np.linalg.norm(nn), 1., atol=1e-4)
-                nn /= np.linalg.norm(nn)
-                nn = apply_trans(curry_dev_dev_t, nn, move=False)
-                trans = np.eye(4)
-                trans[:3, 3] = pos
-                trans[:3, :3] = _normal_orth(nn).T
-                ch['loc'] = _coil_trans_to_loc(trans)
-                ch['coord_frame'] = FIFF.FIFFV_COORD_DEVICE
-            all_chans.append(ch)
+            chanidx = len(all_chans) + 1    # by default, just assume the
+            # channel index in the datafile is in order of the channel
+            # names as we found them in the labels file
+            if chanidx_is_explicit:  # but, if explicitly declared, use
+                # that index number
+                chanidx = int(curry_params.chanidx_in_file["CHAN_IN_FILE"
+                              + CHANTYPES[key]][ind])
+            if chanidx > 0:   # if chanidx was explicitly declared to be ' 0',
+                # it means the channel is not actually saved in the data file
+                # (e.g. the "Ref" channel), so don't add it to our list.
+                # Git issue #8391
```

Comment:
```suggestion
            if chanidx <= 0:   # if chanidx was explicitly declared to be ' 0',
                # it means the channel is not actually saved in the data file
                # (e.g. the "Ref" channel), so don't add it to our list.
                # Git issue #8391
                continue
```
Then you can unindent everything below. It makes the diff and `git blame` nicer, and makes it so there is more PEP8-compatible space to use, and makes it immediately clear it's a short-circuit / ignore branch (don't need to scroll down to look for an `else` to know)

Reply:
Cool, thanks